pulp_service/pulp_service/app/viewsets.py

```diff
@@ -148,19 +151,33 @@ def view_name(cls):
 
 class VulnerabilityReport(ViewSet):
 
-    def list(self, request):
-        queryset = VulnReport.objects.all()
+    def retrieve(self, request):
+        queryset = VulnReport.objects.filter(pulp_domain=get_domain())
         serializer = VulnerabilityReportSerializer(queryset, many=True)
         return Response(serializer.data)
 
-    def get(self, request, uuid):
-        queryset = VulnReport.objects.get(id=uuid)
+    def list(self, request, uuid):
+        try:
+            queryset = VulnReport.objects.get(id=uuid, pulp_domain=get_domain())
+        except ObjectDoesNotExist:
+            raise Http404(f"No VulnerabilityReport found with uuid {uuid}")
         serializer = VulnerabilityReportSerializer(queryset, many=False)
         return Response(serializer.data)
 
-    def post(self, request):
+    def create(self, request):
         serialized_data = ContentScanSerializer(data=request.data)
         serialized_data.is_valid(raise_exception=True)
-        repo_version_pk = serialized_data.data["repo_version"]
-        task = dispatch(check_content, kwargs={"repo_version_pk": repo_version_pk})
+
+        """Dispatch a task to scan the Content Units from a Repository"""
+        if repo_version_pk := serialized_data.data.get("repo_version", None):
+            dispatch_task, kwargs = check_content_from_repo_version, {
+                "repo_version_pk": repo_version_pk
```

Comment:
If you specify a repo-version you should probably use a shared lock on the repository to prevent the task from running on a potentially deleted repo/repo-version.

Reply:
+1 to this